Source/Core/Core/MemoryWatcher.cpp

```diff
@@ -99,13 +146,113 @@ u32 MemoryWatcher::ChasePointer(const std::string& line)
   return value;
 }
 
+bool MemoryWatcher::IsIdentical(blob one, blob two)
+{
+  if(one.size() != two.size())
+    return false;
+
+  for(u32 i = 0; i < one.size(); i++)
+  {
+    if(one[i] != two[i])
+      return false;
+  }
+  return true;
+}
+
+bool MemoryWatcher::ChaseLinkedList(const std::string& address, linked_list &llist)
+{
+  u32 pointer;
+  std::stringstream ss;
+  ss << std::hex << address;
+  ss >> pointer;
+
+  //Follow the first pointer over
+  pointer = Memory::Read_U32(pointer);
+  list_data data;
+
+  bool changed = false;
+  u32 i = 0;
+  while(true)
+  {
+    //Quit on NULL pointer
+    if(pointer == 0)
+    {
+      break;
+    }
+
+    //Grab the data pointer from the current element
+    u32 data_pointer = 0;
+    if(!Memory::CopyFromEmuByHost(&data_pointer, pointer + llist.m_data_pointer_offset, sizeof(data_pointer)))
+    {
+      break;
+    }
+
+    data_pointer = SwapEndianness(data_pointer);
+
+    //Read a blob at the data pointer address. This is the actual data we care about
+    //Read the next element in the list
+    blob chunk;
+    chunk.resize(llist.m_data_struct_len);
+    if(!Memory::CopyFromEmuByHost(chunk.data(), data_pointer, llist.m_data_struct_len))
+    {
+      //If we get an error here, interpret it as a NULL pointer and just quit
+      // Apparently some linked lists in GCN like to terminate pointing at
+      // something that isn't NULL
```

Comment:
Actually, I think I was wrong about that comment. I fixed it in my later commit. 

What was actually happening is that when the game starts up, the memory is uninitialized and it causes us to read in garbage.